azurerm/internal/services/signalr/signalr_service_resource.go

```diff
@@ -390,6 +465,57 @@ func flattenSignalRFeatures(features *[]signalr.Feature) []interface{} {
 	return result
 }
 
+func expandUpstreamSettings(input []interface{}) *signalr.ServerlessUpstreamSettings {
+	upstreamTemplates := make([]signalr.UpstreamTemplate, 0)
+
+	for _, upstreamSetting := range input {
+		setting := upstreamSetting.(map[string]interface{})
+
+		upstreamTemplate := &signalr.UpstreamTemplate{
+			HubPattern:      utils.String(strings.Join(*utils.ExpandStringSlice(setting["hub_pattern"].([]interface{})), ",")),
+			EventPattern:    utils.String(strings.Join(*utils.ExpandStringSlice(setting["event_pattern"].([]interface{})), ",")),
+			CategoryPattern: utils.String(strings.Join(*utils.ExpandStringSlice(setting["category_pattern"].([]interface{})), ",")),
+			URLTemplate:     utils.String(setting["url_template"].(string)),
+		}
+
+		upstreamTemplates = append(upstreamTemplates, *upstreamTemplate)
```

Comment:
since we immediately de-reference it here:

```suggestion
		upstreamTemplates = append(upstreamTemplates, upstreamTemplate)
```